File: src/main/kotlin/com/codingforcookies/betterrecords/client/gui/SlotValid.java
package com.codingforcookies.betterrecords.client.gui;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;

public class SlotValid extends Slot {

    public SlotValid(IInventory iInventory, int index, int x, int y) {
        super(iInventory, index, x, y);
    }

    public boolean isItemValid(ItemStack itemStack) {
        return inventory.isItemValidForSlot(slotNumber, itemStack);
    }
}

File: src/main/kotlin/com/codingforcookies/betterrecords/client/gui/SlotValid.kt
package com.codingforcookies.betterrecords.client.gui

import net.minecraft.inventory.IInventory
import net.minecraft.inventory.Slot
import net.minecraft.item.ItemStack

class SlotValid(inventory: IInventory, index: Int, x: Int, y: Int) : Slot(inventory, index, x, y) {

    override fun isItemValid(itemStack: ItemStack?) =
            inventory.isItemValidForSlot(slotNumber, itemStack)
}
